src/iosrv/init.c

```diff
@@ -589,10 +590,57 @@ server_init(int argc, char *argv[])
 	return rc;
 }
 
+static int
+cleanup_cb(const char *path, const struct stat *sb, int flag,
+	   struct FTW *ftwbuf)
+{
+	int rc;
+
+	if (ftwbuf->level == 0)
+		return 0;
+
+	if (flag == FTW_DP || flag == FTW_D)
+		rc = rmdir(path);
+	else
+		rc = unlink(path);
+	if (rc)
+		D_ERROR("failed to remove %s\n", path);
+	return rc;
+}
+
+static void
+shared_memory_file_cleanup(void)
+{
+	int rc;
+
+	if (dss_nvme_shm_id != DAOS_NVME_SHMID_NONE) {
+		/* unlink spdk shared memory files for this stream */
+		char buffer[NAME_MAX];
+
+		/* remove shared memory dir created byi each daos io searver
+		 * currently in /var/run/dpdk
```

Comment:
If they are in `/var/run` (now -- they used to be in `~` when the ticket was opened) do they actually need to be cleaned up given that `/var/run` is a _tmpfs_ and as such automatically cleaned on a system reboot?

Perhaps we should be cleaning them up between daos server restarts and not just waiting for the node to be rebooted to clean them?

I suppose also, now that this code is written to do the cleanup, there is no harm in being proactive about it.  Although it is more code to maintain (i.e. should the path to the files change again).

Reply:
The consensus of daos control team is good to do the cleanup to avoid files piling up before rebooting,  but is not high priority, so will continue this review process.